packages/react/src/time-input/TimeInput.test.tsx

```diff
@@ -0,0 +1,132 @@
+import '@testing-library/jest-dom';
+import userEvent from '@testing-library/user-event';
+import 'html-validate/jest';
+import { useState } from 'react';
+import { axe, toHaveNoViolations } from 'jest-axe';
+import { cleanup, render, screen } from '../../../../test-utils';
+import { TimeInput, type TimeInputProps, type TimeValue } from './TimeInput';
+
+expect.extend(toHaveNoViolations);
+
+afterEach(cleanup);
+
+type ControlledTimeInputProps = Omit<TimeInputProps, 'label'>;
+
+function renderTimeInput(props?: ControlledTimeInputProps) {
+	return render(<ControlledTimeInput {...props} />);
+}
+
+function ControlledTimeInput({
+	onChange: onChangeProp,
+	value: valueProp,
+	...props
+}: ControlledTimeInputProps) {
+	const [value, setValue] = useState<TimeValue | undefined>(valueProp);
+
+	function onChange(value: TimeValue) {
+		setValue(value);
+		onChangeProp?.(value);
+	}
+
+	return (
+		<TimeInput label="Time" onChange={onChange} value={value} {...props} />
+	);
+}
+
+async function getInput() {
+	const el = await screen.getByRole('textbox');
```

Comment:
`getByRole` doesn't return a promise, so you shouldn't need to `await` this function. Also, if you pass an options object with a `name` property, you add an extra layer of safety as the getter only returns the `textbox` with that accessible name. e.g.:

```ts
const el = screen.getByRole('textbox', { name: 'Time' });
```

Reply:
Good to know. That means loads of other tests are doing it wrong…